src/components/AmountInput.js

```diff
@@ -0,0 +1,43 @@
+import { AmountInputField, AmountLabel, AmountWrapper } from "@/src/components/styles/StyledAmountInput";
+import { useEffect } from "react";
+import { validateInput } from "@/src/utils/validation";
+
+export default function AmountInput({
+                                      id,
+                                      name,
+                                      value,
+                                      type,
+                                      onChange,
+                                      onKeyDown,
+                                      placeholder,
+                                      errorMessage,
+                                      validate,
+                                    }) {
+
+  useEffect(() => {
+    if (validate) {
+      const isValid = validateInput(value, "number");
+      if (!isValid) {
+        document.getElementById(id + "-wrapper").classList.add("has-error");
+      } else {
+        document.getElementById(id + "-wrapper").classList.remove("has-error");
+      }
+    }
+  }, [validate, value, id]);
```

Comment:
You are using here and in the whole app a mix from styling with styled-components and with classes, don't you want to use variables for styled components instead? And do you really need to use useEffect hook here, can't you use a conditional rendering and styles-variables instead of it too?